plugins/Live/Visualizations/VisitorLog.php

```diff
@@ -202,13 +202,54 @@ public static function groupActionsByPageviewId(DataTable $table)
                 }
             }
 
+            foreach ($actionGroups as $idPageView => $actionGroup) {
+                if (!empty($actionGroup['actionsOnPage'])) {
+                    usort($actionGroup['actionsOnPage'], function ($a, $b) {
+                        $fields = array('timestamp', 'title', 'url', 'type', 'pageIdAction', 'goalId', 'timeSpent');
+                        foreach ($fields as $field) {
+                            $sort = self::sortByActionsOnPageColumn($a, $b, $field);
+                            if ($sort !== 0) {
+                                return $sort;
+                            }
+                        }
+
+                        return 0;
+                    });
+                    $actionGroups[$idPageView]['actionsOnPage'] = $actionGroup['actionsOnPage'];
+                }
+            }
+
             // merge action groups that have the same page url/action and no pageviewactions
             $actionGroups = self::mergeRefreshes($actionGroups);
 
             $row->setColumn('actionGroups', $actionGroups);
         }
     }
 
+    public static function sortByActionsOnPageColumn($a, $b, $field)
+    {
+        if (!isset($a[$field]) && !isset($b[$field])) {
+            return 0;
+        }
+        if (!isset($a[$field])) {
+            return -1;
+        }
+        if (!isset($b[$field])) {
+            return -1;
```

Comment:
shouldn't that one be `1` ? 🤔 
```suggestion
            return 1;
```